src/number_theory/wilson.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2022 John Nicol. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Nicol
+-/
+import number_theory.legendre_symbol.gauss_eisenstein_lemmas
+
+/-!
+# Wilson's theorem.
+
+This file contains a proof of Wilson's theorem.
+
+The heavy lifting is mostly done by the previous `wilsons_lemma`,
+but here we also prove the other logical direction.
+
+This could be generalized to similar results about finite abelian groups.
+
+## References
+
+* [Wilson's Theorem](https://en.wikipedia.org/wiki/Wilson%27s_theorem)
+
+-/
+
+open_locale nat
+
+namespace nat
+
+/-- For `n > 1`, `(n-1)!` is congruent to `-1` modulo `n` only if n is prime. --/
+private lemma wilsons_theorem_only_if_direction  (n : ℕ) (h1: 1 < n) :
+(((n - 1)! : zmod n) = -1)  → (prime n):=
+begin
+  intro h,
+  have hp: ((n - 1)! + 1 : zmod n) = 0,
+  rw h,
+  simp,
```

Comment:
```suggestion
  have hp: ((n - 1)! + 1 : zmod n) = 0,
  rw h,
  simp,
```
When you create a new subgoal you should put the proof in `{...}` so that we can see when the subgoal is finished. This applies throughout the file.